shared/poll.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2020 The WPT Dashboard Project. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package shared
+
+import (
+	"net/http"
+	"time"
+)
+
+// KeepMetadataUpdated implements updates to metadataMapCached via simple polling every
+// interval duration.
+func KeepMetadataUpdated(client *http.Client, logger Logger, interval time.Duration) {
+	// Start by waiting polling interval.
+	logger.Infof("Starting Metadata update via polling; waiting polling interval first...")
+	time.Sleep(interval)
+	logger.Infof("Metadata update via polling started")
+
+	for {
+		start := time.Now()
+		err := UpdatedMetadata(client, logger)
+		if err != nil {
+			logger.Errorf("Error fetching Metadata for update: %v", err)
+		}
+
+		wait(start, interval)
```

Comment:
I think this is a bit overly complicated. We can simply `time.Sleep(interval)` again and disregard how much time the download takes.